server/api/contest_handler.py

```diff
@@ -0,0 +1,56 @@
+import json
+from flask import jsonify, Blueprint, request
+from api.models import contest_table, db
+from datetime import date
+contest_handler = Blueprint('contest_new_handler', __name__)
+
+
+@contest_handler.route('/create_new_contest', methods=['POST'])
+def create_contest():
+    if request.method == 'POST':
+
+        title = request.form['title']
+        description = request.form['description']
+        prize_contest = request.form['amount']
+        deadline_date = request.form['deadline_date']
+        created_time = date.today()
+        update_time = date.today()
+        contest_creator = request.json['contestCreator']
+        
+        new_contest = contest_table(title, description, prize_contest, deadline_date, created_time, update_time, contest_creator)
+        db.session.add(new_contest)
+        db.session.commit()
+
+        return jsonify({'successMessage': 'Contest Created!'})
+
+@contest_handler.route('/contests', methods=['GET'])
+def get_all_contests():
+    all_contests = contest_table.query.all()
+    return jsonify(all_contests)
+
+@contest_handler.route('/contest/<contest_id>')
+def get_contest(contest_id):
+    contest = contest_table.query.get(contest_id)
+    return jsonify(contest)  
+
+@contest_handler.route('/update_contest/<contest_id>', methods=['PUT'])
+def update_contest(contest_id):
+    if request.method == 'PUT':
+        contest = contest_table.query.get(contest_id)
+
+        title = request.form['title']
+        description = request.form['description']
+        prize_contest = request.form['amount']
+        deadline_date = request.form['deadline_date']
+        update_time = date.today()
+
+        contest.title = title
```

Comment:
nothing is wrong with this, but just a nit pick for the time being because variables `title`, `description`, `prize_contest`, etc etc, is only used once ever in this function. if you can anticipate it being used more than once, then it's totally fine to keep it. otherwise, it would be better to just call it directly like `contest.title = request.form['title']`. reason for this is cause it gets redundant and if you were to update another attribute, you would have to write it in 2 places rather than 1

up to you if you want to change this! not too strict on this one

Reply:
Okay will change that, wasn't all too sure since I'm not very familiar with all the "best practices"